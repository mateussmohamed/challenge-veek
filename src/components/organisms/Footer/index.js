import React from 'react'
import styled from 'styled-components'

import { Grid, Row, Col } from 'react-styled-flexboxgrid'

import { Icon, Link, Logo } from 'atoms'
import { LinkList } from 'molecules'

const FooterStyled = styled.footer`
  background-color: ${props => props.theme.colors.secondary};
  width: 100%;
  padding: 25px 0;

  @media only screen and (min-width: 0em) {
    margin-top: 100px;
  }
  @media only screen and (min-width: 48em) {
    margin-top: 140px;
  }
  @media only screen and (min-width: 64em) {
    margin-top: 200px;
  }
`
const FooterLogo = styled(Link)`
  @media only screen and (min-width: 0em) {
    text-align: center;
    display: block;
    margin-bottom: 25px;
  }
  @media only screen and (min-width: 48em) {
    margin-bottom: 15px;
  }
`

const SocialLinks = styled(LinkList)`
  flex-direction: row !important;
  li {
    margin-top: 0;
    margin-bottom: 0;
  }
`

const Footer = () => (
  <FooterStyled>
    <Grid style={{ padding: '0 1rem' }}>
      <Row between="xs" middle="sm">
        <Col xs={12} sm={12} md={2}>
          <FooterLogo>
            <Logo color="#fff" />
          </FooterLogo>
        </Col>
        <Col xs={12} sm={4} md={2}>
          <LinkList>
            <Link>Team</Link>
            <Link>About us</Link>
            <Link>Process</Link>
          </LinkList>
        </Col>
        <Col xs={12} sm={4} md={2}>
          <LinkList>
            <Link>Explication</Link>
            <Link>News</Link>
            <Link>Contact</Link>
          </LinkList>
        </Col>
        <Col xs={12} sm={4} md={4}>
          <LinkList direction="row">
            <Link>Privacy Police</Link>
            <Link>Legal Notice</Link>
            <Link>English</Link>
          </LinkList>
        </Col>
        <Col xs={12} sm={12} md={2}>
          <SocialLinks direction="row">
            <Link>
              <Icon name="facebook" />
            </Link>
            <Link>
              <Icon name="twitter" />
            </Link>
            <Link>
              <Icon name="instagram" />
            </Link>
            <Link>
              <Icon name="linkedin" />
            </Link>
          </SocialLinks>
        </Col>
      </Row>
    </Grid>
  </FooterStyled>
)

export default Footer
