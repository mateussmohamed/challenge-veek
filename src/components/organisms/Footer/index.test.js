import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'

import Footer from './index'

describe('[<Footer />]', () => {
  const wrap = (props = {}) => shallow(<Footer {...props} />)

  test('matches to snapshot', () => {
    const wrapper = wrap()

    expect(wrapper).toMatchSnapshot()
  })
})
