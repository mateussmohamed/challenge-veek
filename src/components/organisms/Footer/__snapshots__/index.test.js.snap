// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[<Footer />] matches to snapshot 1`] = `
<Footer__FooterStyled>
  <Grid
    style={
      Object {
        "padding": "0 1rem",
      }
    }
  >
    <Row
      between="xs"
      middle="sm"
    >
      <Col
        md={2}
        sm={12}
        xs={12}
      >
        <Footer__FooterLogo>
          <Logo
            color="#fff"
            height="35px"
            width="90px"
          />
        </Footer__FooterLogo>
      </Col>
      <Col
        md={2}
        sm={4}
        xs={6}
      >
        <LinkList
          direction="column"
        >
          <Link
            color="#fff"
          >
            Team
          </Link>
          <Link
            color="#fff"
          >
            About us
          </Link>
          <Link
            color="#fff"
          >
            Process
          </Link>
        </LinkList>
      </Col>
      <Col
        md={2}
        sm={3}
        xs={6}
      >
        <LinkList
          direction="column"
        >
          <Link
            color="#fff"
          >
            Explication
          </Link>
          <Link
            color="#fff"
          >
            News
          </Link>
          <Link
            color="#fff"
          >
            Contact
          </Link>
        </LinkList>
      </Col>
      <Col
        md={4}
        sm={5}
        xs={12}
      >
        <LinkList
          direction="row"
        >
          <Link
            color="#fff"
          >
            Privacy Police
          </Link>
          <Link
            color="#fff"
          >
            Legal Notice
          </Link>
          <Link
            color="#fff"
          >
            English
          </Link>
        </LinkList>
      </Col>
      <Col
        md={2}
        sm={12}
        xs={12}
      >
        <LinkList
          direction="row"
        >
          <Link
            color="#fff"
          >
            <Icon
              name="facebook"
            />
          </Link>
          <Link
            color="#fff"
          >
            <Icon
              name="twitter"
            />
          </Link>
          <Link
            color="#fff"
          >
            <Icon
              name="instagram"
            />
          </Link>
          <Link
            color="#fff"
          >
            <Icon
              name="linkedin"
            />
          </Link>
        </LinkList>
      </Col>
    </Row>
  </Grid>
</Footer__FooterStyled>
`;
